### PR TITLE
store only is_migrated true in migrated .script files

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes [PLAT-859].

In theory, this change should be safe because we already ignore most of the .script file contents aside from the is_migrated flag when seeding migrated scripts: https://github.com/code-dot-org/code-dot-org/blob/19f9a1c5ca48576f8012336d3b3f8122e6ca99bb/dashboard/app/models/script.rb#L1028-L1030

It takes a bit of inspection of the code just above this to convince yourself that the other accesses to other parts of the parsed script data will be harmless after this change. The code could definitely benefit from being cleaned up so that the migrated script more clearly does not use any of the old script logic. However, I'm erring on the side of changing the least amount possible, since this code will soon be going away once all scripts are migrated.

## side benefit

This fix also fixes the issue mentioned in the the Jira item above about scripts.en.yml changing during seed on staging/test/prod. The reason this was happening was because lesson names were stored in 3 places (script dsl, script json, and scripts.en.yml) but lesson update was failing to update 1 of them (script dsl). Then during seed the scripts.en.yml was getting overwritten with the value from script dsl by this codepath:
* https://github.com/code-dot-org/code-dot-org/blob/19f9a1c5ca48576f8012336d3b3f8122e6ca99bb/dashboard/app/models/script.rb#L1001-L1003
* https://github.com/code-dot-org/code-dot-org/blob/19f9a1c5ca48576f8012336d3b3f8122e6ca99bb/dashboard/app/models/script.rb#L1009

This PR fixes the problem because now no lesson names are pulled out of the script dsl.

## Testing story

Unit tests verify that this does not break serialization of non-migrated scripts, and that the correct output is produced for migrated scripts. 

There are also lots of UI tests making sure that seeding of legacy scripts is working. Based on that, my manual testing focused on migrated scripts.

Here is one set of manual tests I did:
1. go to http://localhost-studio.code.org:3000/s/csp1-2021/edit and press save --> .script file is updated to contain only `is_migrated true`
2. commit those changes to git
3. remove contents from that script via `Script.find_by_name('csp1-2021').lesson_groups.destroy_all`
4. reseed that script via `rake seed:single_script SCRIPT_NAME=csp1-2021`
5. at this point, there may be changes in the database which we cannot see.
6. go to http://localhost-studio.code.org:3000/s/csp1-2021/edit and press save --> no substantial changes to csp1-2021.script_json, thanks in part to #40372

After step 6, I feel reasonably confident that the system is still propagating script data to other environments for migrated scripts. If there were changes to the DB contents in step 5, then I'd expect them to show up in csp1-2021.script_json after step 6.

I am open to other scenarios to test here.

## Deployment strategy

I am planning to follow the normal deploy process for shipping this change. Once this PR reaches levelbuilder, I'll do `write_script_dsl` on each migrated script there to get their .script files into the new desired state. this will help avoid merge conflicts that might happen if I include the changes to the .script files in this PR.

[PLAT-859]: https://codedotorg.atlassian.net/browse/PLAT-859

### CODE REVIEW

**Review by bethanyaconnor — APPROVED** [review]
This make sense to me. Thanks for the thorough write-up of your investigation!

**Comment by dmcavoy** [comment]
Hey @davidsbailey - I have a couple of wonders here. I haven't gotten to write them up but wanted to make sure I said something before you merged. I will try to write up stuff after Josh's presentation this afternoon

**Comment by dmcavoy** [comment]
Alright just getting back to this. Going to post thoughts one by one in case I have to step away. 

My first wonder is about our protection on the script edit page looking for something having been changed. That relies on script dsl. Will it still work? https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/scripts_controller.rb#L121-L128

**Comment by dmcavoy** [comment]
Do we need to update the stuff that is generating the script DSL in the scriptEditorRedux? 

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/levelbuilder/script-editor/scriptEditorRedux.js#L307-L415

**Comment by dmcavoy** [comment]
Do these comments need to be updated?

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/script.rb#L1264-L1270

**Comment by dmcavoy** [comment]
> Do we need to update the stuff that is generating the script DSL in the scriptEditorRedux?
> 
> https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/levelbuilder/script-editor/scriptEditorRedux.js#L307-L415

It looks like the answer to this is no. That we use this serialization to update the database and then we will just not save it because we call write_script_dsl. It feels a little weird to have things which are still relying on script DSL when we are not saving it but I think its ok.

**Comment by davidsbailey** [comment]
> My first wonder is about our protection on the script edit page looking for something having been changed. That relies on script dsl. Will it still work? https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/scripts_controller.rb#L121-L128

It looks like it should, because that data comes from here:
* https://github.com/code-dot-org/code-dot-org/blob/9b139fca6ac5f247516638519f6e6d2f7b764978/dashboard/app/controllers/scripts_controller.rb#L193
* https://github.com/code-dot-org/code-dot-org/blob/9b139fca6ac5f247516638519f6e6d2f7b764978/dashboard/app/controllers/scripts_controller.rb#L112

It's confusing because it uses the variable name `@script_file`, but what it's doing is generating the script DSL based on the contents of the database.

> Do these comments need to be updated? https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/script.rb#L1264-L1270

Updated

> > Do we need to update the stuff that is generating the script DSL in the scriptEditorRedux?
> > https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/levelbuilder/script-editor/scriptEditorRedux.js#L307-L415
> 
> It looks like the answer to this is no. That we use this serialization to update the database and then we will just not save it because we call write_script_dsl. It feels a little weird to have things which are still relying on script DSL when we are not saving it but I think its ok.

yes, we wouldn't leave it like this, but as an intermediate step it seems ok for the dsl to be used in the API but not in the .script file. Check out the Migrate Old Scripts epic for a broader picture of the work that will lead us to getting all scripts migrated. If the info captured in Jira is not a good format, please let me know and I'd be happy to write up some other kind of doc to share.

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -1261,12 +1261,14 @@ def update_text(script_params, script_text, metadata_i18n, general_params)
...
-        # Save in our custom Script DSL format. This is still what we're using currently to sync data
-        # across environments. The CPlat team is working on replacing it a new JSON-based approach.
+        # Save in our custom Script DSL format. This is how we currently sync
+        # data across environments for non-migrated scripts.
         script.write_script_dsl
 
-        # Also save in JSON format for "new seeding". This has not been launched yet for most scripts, but as part of
-        # pre-launch testing, we'll start generating these files in addition to the old .script files.
+        # Also save in JSON format for "new seeding". This is how we currently
+        # sync data across environments for non-migrated scripts. As part of
```
Comment (dmcavoy) [inline dashboard/app/models/script.rb]: nit: migrated scripts

**Review by dmcavoy — APPROVED** [review]
Thanks for waiting on my comments and then responding to them!

**Pull request merged**